strum_tests/tests/enum_discriminants.rs

```diff
@@ -375,3 +375,16 @@ fn non_exhaustive_enum() {
         _ => unreachable!(),
     }
 }
+
+#[allow(dead_code)]
+#[derive(Debug, Eq, PartialEq, EnumDiscriminants)]
+#[strum_discriminants(derive(EnumIter))]
+enum Empty {}
+
+#[test]
+fn empty_test() {
+    let discriminants = EmptyDiscriminants::iter().collect::<Vec<EmptyDiscriminants>>();
+    let expected: Vec<EmptyDiscriminants> = vec![];
+
+    assert_eq!(expected, discriminants);
```

Comment:
👑 If the code stays like this, then the vec type could be elided in the first line:

```suggestion
    let discriminants = EmptyDiscriminants::iter().collect::<Vec<_>>();
    let expected: Vec<EmptyDiscriminants> = vec![];

    assert_eq!(expected, discriminants);
```

However, we could potentially skip creating `Vec`s in the first place:

```suggestion
    let num_discriminants = EmptyDiscriminants::iter().count();
    assert_eq!(num_discriminants, 0);
```